### PR TITLE
Improvement: new events management

### LINKED ISSUE #217: `hasError` function missing when updating/adding/saving attributes in Component Detail Panel
**Describe the bug**

An error occurs when we update some attributes in the Component Detail Panel. It is linked to `hasError` function that comes from `ComponentAttribute` class. Currently, when we update/save/add an attribute, we rewrite the entire object and therefore we lose all functions related to that class, like `hasError` function causing this bug (+ error in console). 
Also same error occurs when we click on 'save' for component that already has attributes (see examples below)

**To Reproduce**

1. Create a new project and a new model.
2. Click on `aws ami` component from `terrator-plugin` in the Model view
3. Click on `Test application` component from `Templates` in the Model view
4. Select `aws ami` and click on `+ Add attribute` button on Component Detail Panel
5. Click on `SAVE` and see error in screenshot

**Screenshots**

_Select component :_
aws_ami

_After adding attribute and clicking save:_
aws_ami_after

Another example with a template component
_Select component  :_
security_group

_After clicking save :_
security_group_after

**Application (please complete the following information):**

- leto version [1.0.0]
- leto-modelizer-plugin-core [0.15.0]
- terrator-plugin : branch `improvement/emit_events` 
- ("terrator-plugin": "github:ditrit/terrator-plugin#improvement/emit_events")

### PR DESCRIPTION
**Pull request by justinehell**
Update event management with upgrade of plugin-core version 0.15.2
Also fixed #217

### CODE REVIEW

File: src/components/ModelizerDrawView.vue [hunk 1]
```diff
@@ -76,15 +77,30 @@ const defaultFolder = ref(process.env.MODELS_DEFAULT_FOLDER !== ''
   : '');
 
 /**
- * Render the model.
+ * On 'Drawer' event type, call renderConfiguration if action is 'move',
+ * otherwise call renderModel.
+ * @param {Object} eventManager - Object containing event information.
```
Comment (k-y0u) [inline src/components/ModelizerDrawView.vue:82]: update description

File: src/components/ModelizerDrawView.vue [hunk 2]
```diff
@@ -76,15 +77,30 @@ const defaultFolder = ref(process.env.MODELS_DEFAULT_FOLDER !== ''
...
  * @return {Promise<void>} Promise with nothing on success otherwise an error.
  */
-async function renderModelComponents() {
-  await renderModel(
-    props.projectName,
-    `${defaultFolder.value}${query.value.path}`,
-    data.plugin,
-  );
+async function onDefaultEvent({ event }) {
+  const renderModelActionList = ['update', 'delete', 'add'];
```
Comment (k-y0u) [inline src/components/ModelizerDrawView.vue:87]: update name

File: src/components/drawer/ComponentDetailPanel.vue [hunk 3]
```diff
@@ -245,20 +245,27 @@ function reset() {
...
  */
-function onEdit({ id }) {
-  isVisible.value = true;
+function setLocalValues({ event }) {
+  if (!event.components || !event.components[0]) {
+    return;
+  }
 
-  originalComponent.value = props.plugin.data.getComponentById(id);
+  if (event.type === 'Drawer' && event.action === 'select') {
```
Comment (Zorin95670) [inline src/components/drawer/ComponentDetailPanel.vue:257]: Rename commit `Update ComponentDetailPanel` to be more precise

File: src/composables/PluginManager.js [hunk 4]
```diff
@@ -159,6 +159,26 @@ export async function renderModel(projectId, modelPath, plugin) {
...
+  const config = new FileInput({
+    path: `${modelPath}/leto-modelizer.config.json`,
+    content: '{}',
+  });
+
+  // TODO : replace by appropriate function when it's done in plugin-core
+  // eslint-disable-next-line no-underscore-dangle
+  plugin.__renderer.renderConfiguration(config);
+
+  return writeProjectFile(projectId, config);
```
Comment (k-y0u) [inline src/composables/PluginManager.js:179]: return -> await

File: src/components/drawer/ComponentDetailPanel.vue [hunk 5]
```diff
@@ -245,20 +245,27 @@ function reset() {
 }
 
 /**
- * Display panel and init local values according to the received id.
- * @param {Object} id - Id of the component to display.
+ * On 'Drawer' event type and 'select' action, display panel and init local values.
+ * @param {Object} eventManager - Object containing event information.
```
Comment (k-y0u) [inline src/components/drawer/ComponentDetailPanel.vue:249]: same here event + plugin

File: src/components/ModelizerDrawView.vue (hunk 1 above)
Comment (justinehell) [reply src/components/ModelizerDrawView.vue:82]: done

File: src/components/drawer/ComponentDetailPanel.vue [hunk 6]
```diff
@@ -137,15 +137,15 @@ function sanitizeAttributes(attributes) {
   return attributes.reduce((acc, attribute) => {
     if (attribute.value && attribute.value !== '') {
       if (attribute.type !== 'Object') {
-        acc.push(attribute);
+        acc.push(new ComponentAttribute(attribute));
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/drawer/ComponentDetailPanel.vue:140]: done

File: src/composables/PluginManager.js (hunk 4 above)
Comment (justinehell) [reply src/composables/PluginManager.js:179]: done

File: src/components/drawer/ComponentDetailPanel.vue (hunk 5 above)
Comment (justinehell) [reply src/components/drawer/ComponentDetailPanel.vue:249]: done

**Review by k-y0u — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**